wordpress-mobile/WordPress-Android
# Issue/5325 setting newly added self hosted site as selected

Fixes #5325 

The original issue and related wpcom PR mentioned utilizing `onSiteChanged` event for selecting a newly added self-hosted site, but after doing some digging, I decided to use `onProfileFetched` event which returns us a site model with valid ID, which we can use to mark site as selected. I believe this is the least intrusive approach (albeit a tiny bit hacky), that utilizes existing site creation flow, but since I'm not that familiar with Login flow, I might be missing something obvious (I tested this approach, and everything worked as expected).

To test:
- While logged in to the app add a self-hosted site from Site Picker.
- When you return to the MySite screen, notice that the added self-hosted site is selected.
